aspnetcore/fundamentals/app-state.md

```diff
@@ -1,188 +1,274 @@
+HTTP is a stateless protocol. Without taking additional steps, HTTP requests are independent messages that don't retain user values or app state. This article describes several approaches to preserve user data and app state between requests.
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/app-state/samples) ([how to download](xref:tutorials/index#how-to-download-a-sample))
+
+## State management
+
+State can be stored using several approaches. Each approach is described later in this topic.
+
+| Storage approach | Storage mechanism |
+| ---------------- | ----------------- |
+| [Cookies](#cookies) | HTTP cookies (may include data stored using server-side app code) |
+| [Session state](#session-state) | HTTP cookies and server-side app code |
+| [TempData](#tempdata) | HTTP cookies or session state |
+| [Query strings](#query-strings) | HTTP query strings |
+| [Hidden fields](#hidden-fields) | HTTP form fields |
+| [HttpContext.Items](#httpcontextitems) | Server-side app code |
+| [Cache](#cache) | Server-side app code |
+| [Dependency Injection](#dependency-injection) | Server-side app code |
+
+## Cookies
+
+Cookies store data across requests. Because cookies are sent with every request, their size should be kept to a minimum. Ideally, only an identifier should be stored in a cookie with the data stored by the app. Most browsers restrict cookie size to 4096 bytes, and only a limited number of cookies are available for each domain.
+
+Because cookies are subject to tampering, they must be validated by the app. Cookies can be deleted by users and expire on clients. However, cookies are generally the most durable form of data persistence on the client.
+
+Cookies are often used for personalization, where content is customized for a known user. The user is only identified and not authenticated in most cases. The cookie can store the user's name, account name, or unique user ID (such as a GUID). You can then use the cookie to access the user's personalized settings, such as their preferred website background color.
+
+Be mindful of the [European Union General Data Protection Regulations (GDPR)](https://ec.europa.eu/info/law/law-topic/data-protection) when issuing cookies and dealing with privacy concerns. For more information, see [General Data Protection Regulation (GDPR) support in ASP.NET Core](xref:security/gdpr).
 
 ## Session state
 
-Session state is a feature in ASP.NET Core that you can use to save and store user data while the user browses your web app. Consisting of a dictionary or hash table on the server, session state persists data across requests from a browser. The session data is backed by a cache.
+Session state is an ASP.NET Core scenario for storage of user data while the user browses a web app. Session state uses a dictionary or hash table maintained by the app to persist data across requests from a client. The session data is backed by a cache.
```

Comment:
~~dictionary or hash table~~ store

Reply:
is backed by a cache and considered ephemeral data, the site should continue to function without it.

Reply:
I think by "it" at the end you mean "the cache," correct?

Reply:
without the session data.

Reply:
Gotcha ... that's the problem with pronouns ... sometimes easy to run amok on which noun they reference. :smile: